CONTRIBUTING.rst

```diff
@@ -478,6 +478,13 @@ Static methods
   higher order concerns than the one about whether a method's body currently
   references self or not.
 
+Assert Messaging
```

Comment:
This also applies to raising exceptions in general. Whenever possible `raise FooException('message', x)` should be used. There should be an example for each of `assert`, `require` and `raise ...`

```suggestion
Error Messages
```

The assert is also special in that it shouldn't even have an error message. That should be mentioned, too,